test/extended/authorization/authorization.go

```diff
@@ -1241,3 +1242,93 @@ var _ = g.Describe("[sig-auth][Feature:OpenShiftAuthorization] authorization", f
 		})
 	})
 })
+var _ = g.Describe("[sig-auth][Feature:OpenShiftAuthorization] ImageRegistry access", func() {
+    defer g.GinkgoRecover()
+    oc := exutil.NewCLI("bootstrap-policy")
+
+    g.Context("", func() {
+        g.Describe("PublicImageAccessWithBasicAuthShouldSucceed", func() {
+            g.It("should succeed [apigroup:image.openshift.io]", func() {
+
+                g.By("Enable default route for the image registry")
+                err := oc.AsAdmin().WithoutNamespace().Run("patch").Args(
+                    "configs.imageregistry.operator.openshift.io/cluster",
+                    "--type=merge",
+                    "-patch", `{"spec": {"defaultRoute": true}}`,
+                ).Execute()
+                o.Expect(err).NotTo(o.HaveOccurred())
+
+                g.By("Wait for the image registry to become available")
+                available := isImageRegistryAvailable(oc)
+                o.Expect(available).To(o.BeTrue())
+
+                g.By("Grant public access to the openshift namespace")
+                defer oc.AsAdmin().WithoutNamespace().Run("policy").Args(
+                    "remove-role-from-group",
+                    "system:image-puller", "system:unauthenticated",
+                    "--namespace", "openshift",
+                ).Execute()
+                output, err := oc.AsAdmin().WithoutNamespace().Run("policy").Args(
+                    "add-role-to-group", "system:image-puller", "system:unauthenticated",
+                    "--namespace", "openshift",
+                ).Output()
+                o.Expect(err).NotTo(o.HaveOccurred())
+                o.Expect(output).To(o.ContainSubstring(
+                    "clusterrole.rbac.authorization.k8s.io/system:image-puller added: \"system:unauthenticated\"",
+                ))
+
+                g.By("Fetch image metadata")
+                output, err = oc.AsAdmin().Run("image").Args(
+                    "info", "--insecure",
+                    "registry.openshift-image-registry.svc.cluster.local/openshift/tools:latest",
+                    "--show-multiarch",
+                ).Output()
+                o.Expect(err).NotTo(o.HaveOccurred())
+                o.Expect(output).NotTo(o.ContainSubstring("error: unauthorized: authentication required"))
+                o.Expect(output).NotTo(o.ContainSubstring("Unable to connect to the server: no basic auth credentials"))
+                o.Expect(output).To(o.ContainSubstring(
+                    "registry.openshift-image-registry.svc.cluster.local/openshift/tools:latest",
+                ))
+            })
+        })
+    })
+})
+
+func isImageRegistryAvailable(oc *exutil.CLI) bool {
+    output, err := oc.AsAdmin().Run("get").Args(
+        "co", "image-registry",
+        "-o=jsonpath={.status.conditions[?(@.type=='Available')].status}",
+    ).Output()
+    o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
TL;DR: have a look at this function: https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/internalimageregistry/management_state.go#L12 - I think you might want to copy/paste it here or at least get some inspiration for how to do this check.

---

You might need to adapt this to work in the following situations:

1. when running in a cluster where the image registry operator is installed, but `.spec.managementState` is set to `Removed`: here I'm uncertain if the Available condition will be set with status true... even if the Available condition would be set to true, it would be misleading for this function to return true, since the image registry would be Removed.
2. when running in a cluster with the `ImageRegistry` capability disabled: in this case, this assertion will definitely end up in an error - I suspect the image registry Cluster Operator isn't installed at all in a cluster where the ImageRegistry capability is disabled.

`.spec.managementState` will be set to `Removed` automatically by the image registry operator when its running in a cluster without object storage available, like metal or vsphere installations for example. If the image registry operator is Removed then you'll want to skip your tests. You can detect this by looking for a Removed condition.

In both these cases you probably want your function to return `false` without it being treated as an error.
You probably want to treat both these cases as correct behavior (instead of an error), and return `false` since the registry will not be available.

[This function](https://github.com/openshift/cluster-openshift-controller-manager-operator/blob/master/pkg/operator/internalimageregistry/management_state.go#L12) implements this check, you might want to use it as inspiration or even copy/paste here and use it instead.